llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -463,6 +485,20 @@ static string_vector saveDeviceImageProperty(
           {"isEsimdImage", true});
     }
 
+    if (ImgPSInfo.IsAssertEnabled) {
+      Module *M = ResultModules[I].get();
+      std::vector<Function *> SyclKernels;
+      for (auto &F : M->functions()) {
+        if (F.getCallingConv() == CallingConv::SPIR_KERNEL) {
```

Comment:
what about external functions? I believe in dynamic linkage case we'll have device libraries which went through post-link (unlike static linkage). Also, RT which links to a dynamic library with "assert used" set, should probably  treat it as "sticky" and mark linked binary image as such too. Suggest to add a TODO for now.